crates/re_edit_ui/src/datatype_editors/enum_combobox.rs

```diff
@@ -1,30 +1,52 @@
+use re_ui::UiExt;
+
 use crate::response_utils::response_with_changes_of_inner;
 
-pub fn edit_enum<Value: PartialEq + std::fmt::Display + Copy>(
+pub fn edit_enum<EnumT: re_types_core::reflection::Enum>(
     ui: &mut egui::Ui,
     id_source: &str,
-    value: &mut Value,
-    variants: &[Value],
+    current_value: &mut EnumT,
 ) -> egui::Response {
     if ui.is_enabled() {
-        response_with_changes_of_inner(
-            egui::ComboBox::from_id_source(id_source)
-                .selected_text(format!("{value}"))
-                .show_ui(ui, |ui| {
-                    let mut iter = variants.iter().copied();
-                    let Some(first) = iter.next() else {
-                        return ui.label("<no variants>");
-                    };
+        let prev_selected_value = *current_value;
+
+        let mut combobox_response = egui::ComboBox::from_id_source(id_source)
```

Comment:
id source could now be generated from reflection as well which should make most callsites one-liners :)